rs/bitcoin/checker/src/main.rs

```diff
@@ -515,6 +524,24 @@ pub async fn check_transaction_inputs(txid: Txid) -> CheckTransactionResponse {
     }
 }
 
+/// Check the input addresses of a transaction given its txid without performing any HTTP outcalls.
+/// A `Pass` or `Fail` status will only be returned if the txid and all of its inputs were already
+/// fetched and available in heap memory. Otherwise, `HttpOutcallRequired` is returned.
+pub async fn check_fetched_transaction_inputs(txid: Txid) -> CheckTransactionQueryResponse {
+    match BtcCheckerCanisterEnv.config().check_mode {
+        CheckMode::AcceptAll => CheckTransactionQueryResponse::Passed,
+        CheckMode::RejectAll => CheckTransactionQueryResponse::Failed(Vec::new()),
+        CheckMode::Normal => {
+            if let Some(FetchTxStatus::Fetched(fetched)) = state::get_fetch_status(txid) {
+                if let Some(response) = check_no_input_address_is_blocked(&fetched) {
+                    return response.into();
+                }
+            }
+            CheckTransactionQueryStatus::HttpOutcallRequired.into()
+        }
```

Comment:
nit: you could have everything under a single match statement
```suggestion
        CheckMode::Normal => match state::get_fetch_status(txid) {
            Some(FetchTxStatus::Fetched(fetched)) => {
                check_for_blocked_input_addresses(&fetched).into()
            }
            Some(
                FetchTxStatus::PendingOutcall
                | FetchTxStatus::PendingRetry { .. }
                | FetchTxStatus::Error(_),
            )
            | None => CheckTransactionQueryResponse::Unknown,
        },
```

Reply:
Nice! I didn't realize I could use the `|` operator _inside_ `Some(...)`. I thought I had to do something like `Some(A) | Some(B) | ... | None` which didn't look so great...  Always happy to learn some new Rust tricks!